version: '3.7'

services:
  nginx-lua:
    image: fabiocicerchia/nginx-lua:${NGINX_LUA_TAG}
    restart: unless-stopped
    volumes:
      - ./configs/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./configs/handler.lua:/etc/nginx/handler.lua:ro
      - ./episodes:/var/episodes:ro
      - ./static:/usr/share/nginx/html/:ro
    healthcheck:
      test: ['CMD', 'curl --fail http://localhost/ || exit 1']
      interval: 30s
      timeout: 3s
      retries: 3
    networks:
      - internal

  php:
    restart: unless-stopped
    image: php:${PHP_TAG}
    networks:
      - internal
    volumes:
      - ./episodes:/var/episodes:ro
      - ./static:/usr/share/nginx/html/:ro
  caddy:
    image: caddy:${CADDY_TAG}
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
      - "443:443/udp"
    networks:
      - internal
    volumes:
      - ./configs/Caddyfile:/etc/caddy/Caddyfile
      - caddy_data:/data
      - caddy_config:/config

  prometheus:
    image: prom/prometheus:v2.43.0
    volumes:
      - ./configs/prometheus.yml:/etc/prometheus/prometheus.yml:ro
      - prometheus_data:/prometheus
    command:
      - '--config.file=/etc/prometheus/prometheus.yml'
      - '--storage.tsdb.path=/prometheus'
      - '--web.console.libraries=/usr/share/prometheus/console_libraries'
      - '--web.console.templates=/usr/share/prometheus/consoles'
    networks:
      - internal
    restart: unless-stopped

  node-exporter:
    image: prom/node-exporter:v1.5.0
    privileged: true
    command: 
      - '--path.rootfs='
      - '--path.procfs=/proc' 
      - '--path.sysfs=/sys'
      - --collector.filesystem.ignored-mount-points
      - "^/(sys|proc|dev|host|etc|rootfs/var/lib/docker/containers|rootfs/var/lib/docker/overlay2|rootfs/run/docker/netns|rootfs/var/lib/docker/aufs)($$|/)"
    networks:
      - internal
    restart: unless-stopped
    deploy:
      mode: global

volumes:
  caddy_data:
    external: true
  caddy_config:
  prometheus_data:

networks:
  internal:
    driver: bridge